syncer/syncer.go

```diff
@@ -742,15 +742,19 @@ func (s *Syncer) addJob(job *job) error {
 	case ddl:
 		// only save checkpoint for DDL and XID (see above)
 		s.saveGlobalPoint(job.location)
-		if len(job.sourceSchema) > 0 {
-			s.saveTablePoint(job.sourceSchema, job.sourceTable, job.location)
+		for i := range job.sourceSchemas {
+			if len(job.sourceSchemas[i]) > 0 {
+				s.saveTablePoint(job.sourceSchemas[i], job.sourceTables[i], job.location)
+			}
 		}
 		// reset sharding group after checkpoint saved
 		s.resetShardingGroup(job.targetSchema, job.targetTable)
 	case insert, update, del:
 		// save job's current pos for DML events
-		if len(job.sourceSchema) > 0 {
-			s.saveTablePoint(job.sourceSchema, job.sourceTable, job.currentLocation)
+		for i := range job.sourceSchemas {
+			if len(job.sourceSchemas[i]) > 0 {
+				s.saveTablePoint(job.sourceSchemas[i], job.sourceTables[i], job.location)
```

Comment:
Should we use `job. location` or `job.currentLocation` here?

Reply:
updated in https://github.com/pingcap/dm/pull/587/commits/f1837f8d8dd5d9aba2c9442fc5f5f9326745a019